core/drand_proxy.go

```diff
@@ -14,13 +17,113 @@ type drandProxy struct {
 	r drand.PublicServer
 }
 
+// streamProxy directly relays mesages of the PublicRandResponse stream.
+type streamProxy struct {
+	ctx      context.Context
+	cancel   context.CancelFunc
+	incoming chan *drand.PublicRandResponse
+	outgoing chan *drand.PublicRandResponse
+}
+
+func newStreamProxy(ctx context.Context) *streamProxy {
+	ctx, cancel := context.WithCancel(ctx)
+	s := streamProxy{
+		ctx:      ctx,
+		cancel:   cancel,
+		incoming: make(chan *drand.PublicRandResponse, 0),
```

Comment:
Why not making it a buffered chan ? 
In the grpc world,  pushing a drand public response is not blocking by default, grpc handles it underneath. Is it happenning here as well ?

Reply:
what do i get from the buffered channel besides an additional buffer? 🤔

Reply:
the only way to guarantee this is non-blocking is to drop messages in the `loop()` if one of the two sides has filled up its queue. i can do that.

Reply:
er - i actually already do do that. the write attempt loop makes has a `default:` case

Reply:
Sounds good - i just wanted to make sure the call is not blocking